internal/service/common/device.go

```diff
@@ -282,31 +282,13 @@ func updateServerSideApplicationStatus(device *api.Device) bool {
 		device.Status.ApplicationsSummary.Info = lo.ToPtr(DeviceStatusInfoRebooting)
 		return device.Status.ApplicationsSummary.Status != lastApplicationSummaryStatus
 	}
-
-	appErrors := []string{}
-	appDegradations := []string{}
-
-	for _, app := range device.Status.Applications {
-		switch app.Status {
-		case api.ApplicationStatusError:
-			appErrors = append(appErrors, fmt.Sprintf("%s is in status %s", app.Name, string(app.Status)))
-		case api.ApplicationStatusPreparing, api.ApplicationStatusStarting:
-			appDegradations = append(appDegradations, fmt.Sprintf("%s is in status %s", app.Name, string(app.Status)))
-		}
-	}
-
-	switch {
-	case len(device.Status.Applications) == 0:
+	if len(device.Status.Applications) == 0 {
```

Comment:
note: removed server side recalculating when apps are available.

An agent with an application that has multiple workloads may have one error out, and one remain running. Currently the device reports that the application is running, but the application summary as "degraded". This server side logic was recalculating the summary (incorrectly) and stating everything was healthy. Now the server side calculations only happen in some cases.